### PR TITLE
[RestorePoints] Fixing instanceView

### PR DESCRIPTION
**Pull request by prchin**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by grizzlytheodore** [comment]
breaking change present 
@prchin, will you provide breaking change justifications for @JeffreyRichter ? 
Please review/approve Jeffrey

**Comment by JeffreyRichter** [comment]
> breaking change present @prchin, will you provide breaking change justifications for @JeffreyRichter ? Please review/approve Jeffrey

When I look here: https://github.com/Azure/azure-rest-api-specs/pull/18592/checks?check_run_id=6068271845
I can't see what the breaks are so I can't approve this until I know the impact.

**Comment by grizzlytheodore** [comment]
@ArcturusZhang the sdk-for-net CI check has been running forever, how can I make it restart? I've closed PR reopend, and tried / azp run

File: specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-01/ComputeRP/restorePoint.json [hunk 1]
```diff
@@ -833,11 +830,13 @@
...
         "status": {
-          "type": "object",
-          "items": {
-            "$ref": "./computeRPCommon.json#/definitions/InstanceViewStatus"
-          },
+          "$ref": "./computeRPCommon.json#/definitions/InstanceViewStatus",
           "description": "The resource status information."
+        },
+        "completionPercent": {
+          "type": "integer",
```
Comment (shahja1) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-01/ComputeRP/restorePoint.json:837]: Is there a way to represent this as nullable int or optional property?

**Review by shahja1 — APPROVED** [review]

**Comment by grizzlytheodore** [comment]
> > breaking change present @prchin, will you provide breaking change justifications for @JeffreyRichter ? Please review/approve Jeffrey
> 
> When I look here: https://github.com/Azure/azure-rest-api-specs/pull/18592/checks?check_run_id=6068271845 I can't see what the breaks are so I can't approve this until I know the impact.

@JeffreyRichter  the check finished. Its showing no breaking change

**Comment by mentat9** [comment]
> > > breaking change present @prchin, will you provide breaking change justifications for @JeffreyRichter ? Please review/approve Jeffrey
> > 
> > 
> > When I look here: https://github.com/Azure/azure-rest-api-specs/pull/18592/checks?check_run_id=6068271845 I can't see what the breaks are so I can't approve this until I know the impact.
> 
> @JeffreyRichter the check finished. Its showing no breaking change

This appears to be a breaking change, since it's changing properties that shipped in API version 2021-11-01 by removing an "items" wrapper from two properties: https://github.com/Azure/azure-rest-api-specs/blob/d8700f40abc50e363891802af12261eae32b6abc/specification/compute/resource-manager/Microsoft.Compute/stable/2021-11-01/compute.json#L9957, and https://github.com/Azure/azure-rest-api-specs/blob/d8700f40abc50e363891802af12261eae32b6abc/specification/compute/resource-manager/Microsoft.Compute/stable/2021-11-01/compute.json#L9970

I don't have any ARM feedback, but I'll wait for the breaking change to be resolved before signing off.

**Comment by JeffreyRichter** [comment]
It would be great to not make the breaking changes.
And, I think this needs to be looked into and fixed: https://github.com/Azure/azure-rest-api-specs/pull/18592/checks?check_run_id=6071269306

File: specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-01/ComputeRP/restorePoint.json (hunk 1 above)
Comment (prchin) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-01/ComputeRP/restorePoint.json:837]: In the generated Sdk we can see this is generated as an nullable int. I had taken this from pre-exisiting diskSizeInGB which was also nullable

Comment (prchin) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-01/ComputeRP/restorePoint.json:837]: https://github.com/Azure/azure-sdk-for-net/pull/28188/files

**Comment by dikethir** [comment]
Hi @JeffreyRichter , restore points are in preview and are primarily used by ISV partners to build BCDR solutions. Currently we have only two partners who are working on implementing BCDR solution using VM restore points and they are already made aware of this change and it does not affect their implementation. We want to fix this as soon as possible before more partners/customers start using VM restore points. So even though it is a breaking change currently it does not affect customers/partners so can we please consider making the change.

**Comment by JeffreyRichter** [comment]
OK, I approved it but I don't feel easy about it. You say it is in preview but your api-version does not have the -preview suffix on it. So, to me and to customers, this looks like it has GA'd and is not preview. If you are truely in preview (and always have been), then you should not merge this PR and instead change the api-version to add the -preview suffix and make a new PR.

**Comment by grizzlytheodore** [comment]
@dikethir approved from jeffrey. please approve

**Comment by grizzlytheodore** [comment]
@JeffreyRichter  currently compute does not utilize '-preview' for API version names, I will bring this up for discussion and figure out the reasons and solutions.

**Comment by dikethir** [comment]
> @dikethir approved from jeffrey. please approve

Approved from myside as well. Thank you!

**Pull request merged**